server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -294,14 +307,7 @@ public boolean isForceExecution() {
                                     }
 
                                     @Override
-                                    public void onFailure(Exception e) {
-                                        sendErrorResponse(reg.getAction(), transportChannel, e);
-                                    }
-
-                                    @Override
-                                    public void onAfter() {
-                                        request.decRef();
-                                    }
+                                    public void onFailure(Exception e) {}
```

Comment:
This is not ok, the executor might directly call `onFailure` when it rejects the runnable because its queue is full. We have to go through the normal error handling mechanism here in this callback if we are using `AbstractRunnable`.